Added detailed pages for each backend API and completed documentatio…

HarshitaShroff opened the pull request:
Updates to the Ontrack documentation website 

# Description

Changes include detailed pages for each backend doubtfire API and completed documentation for API_Set1

- activity_types   
- auth
- campuses
- csv
- settings
- students
- teaching_periods


Fixes # (issue)

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [X] Documentation (update or new)

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration_

## Testing Checklist

- [x] Tested in latest Chrome
- [x] Tested in latest Safari
- [x] Tested in latest Firefox

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from ... on the Pull Request

DudeUnleashed commented:
![image](https://github.com/thoth-tech/doubtfire-astro/assets/51482783/234e3ed3-b5e6-49fd-b4c2-b5a130c036ed)
The underscore for the activity_types overviews are missing, otherwise this section i feel is complete

Missing '/auth/jwt' endpoints under the auth category

Campuses is complete

![image](https://github.com/thoth-tech/doubtfire-astro/assets/51482783/90caf454-7d9f-4edd-a3dc-4604cc8797a0)
Username could have a better description - suggestion: "Username of the user"
Otherwise everything seems covered

Settings is complete
Students is complete

![image](https://github.com/thoth-tech/doubtfire-astro/assets/51482783/e67a43aa-9d3f-4f21-8625-cf690101187f)
Delete teaching period has a markdown error for the parameters table
Otherwise seems correct

HarshitaShroff commented:
I have addressed the above feedback and suggestion.

Activity types: added the underscore for the activity_types overviews.
Auth: Added  '/auth/jwt' endpoints under the auth category
Csv: updated the username description
Teaching_period: Updated the format for DELETE operation

DudeUnleashed commented:
Re-reviewed the documentation and all seems correct, changes listed previously addressed.
